### ISSUE TITLE
Loss calculation issue

### ISSUE THREAD

**Issue by wonagram** [minhtriet/graddft-qnn#30]
```
            avg_cost = sum(cost_values) / len(batch)
        aggregated_train_loss += avg_cost
    train_loss = np.sqrt(aggregated_train_loss / len(train_ds))
    logging.info(f"RMS train loss: {train_loss}")
    train_losses.append(train_loss)
```

We already compute `avg_cost = sum(cost_values) / len(batch) ` for each batch.

So we should divide by number of batches instead of `len(train_ds)`

Or we change `avg_cost = sum(cost_values)`

```
            avg_cost = sum(cost_values)
        aggregated_train_loss += avg_cost
    train_loss = np.sqrt(aggregated_train_loss / len(train_ds))
    logging.info(f"RMS train loss: {train_loss}")
    train_losses.append(train_loss)
```

And `len(batch)` is not the number of data points. `len(batch["symbols"])` represents that.

**Comment by minhtriet** [comment]
`len(batch)` is the number of sample inside the batch. I think you are right here. However it affects the report number only, and not the training as far as I can see
